packages/flutter_tools/templates/plugin/lib/projectName.dart.tmpl

```diff
@@ -13,8 +13,8 @@ class {{pluginDartClass}} {
   static const MethodChannel _channel =
       const MethodChannel('{{projectName}}');
 
-  static Future<String> get platformVersion async {
-    final String version = await _channel.invokeMethod('getPlatformVersion');
+  static Future<String?> get platformVersion async {
```

Comment:
Should this actually be allowed to return null? Or is `null` always an error case (e.g. plugin wasn't available).

Reply:
If I'm reading this correctly, then `MethodChannel` can never return null, but `OptionalMethodChannel` can.
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/services/platform_channel.dart#L330

Not sure how to restructure that code given the shared superclass though?

Reply:
This seems to work:

```diff
diff --git a/packages/flutter/lib/src/services/platform_channel.dart b/packages/flutter/lib/src/services/platform_channel.dart
index c9bad0e210..bd07c68829 100644
--- a/packages/flutter/lib/src/services/platform_channel.dart
+++ b/packages/flutter/lib/src/services/platform_channel.dart
@@ -327,8 +327,8 @@ class MethodChannel {
   ///  * <https://api.flutter.dev/javadoc/io/flutter/plugin/common/MethodCall.html>
   ///    for how to access method call arguments on Android.
   @optionalTypeArgs
-  Future<T?> invokeMethod<T>(String method, [ dynamic arguments ]) {
-    return _invokeMethod<T>(method, missingOk: false, arguments: arguments);
+  Future<T> invokeMethod<T>(String method, [ dynamic arguments ]) {
+    return _invokeMethod<T>(method, missingOk: false, arguments: arguments) as T;
   }
```

Who know this code best and can comment?

Reply:
@goderbauer WDYT?

Reply:
Looks like changing the interface requires a larger refactoring...